hazelcast/src/main/java/com/hazelcast/query/impl/SortedIndexStore.java

```diff
@@ -178,10 +167,109 @@ public void clear() {
         }
     }
 
+    /**
+     * Adds entry to the given index map without copying it.
+     *
+     * @see IndexCopyBehavior
+     */
+    private class AddFunctor implements IndexFunctor<Comparable, QueryableEntry> {
```

Comment:
can we comment that `AddFunctor` must be accessed in a thread safe manner?

Reply:
Actually it doesn't have to. Since it assumes that the map is thread-safe (it uses the concurrent map in this case)

Reply:
But otherwise some entries can be missed from index. Using concurrent-map doesn't help here. Below code is from `AddFunctor` and `records` can be initialized by multiple threads?
```
 Map<Data, QueryableEntry> records = recordMap.get(attribute);
                if (records == null) {
                    records = new ConcurrentHashMap<Data, QueryableEntry>(1, LOAD_FACTOR, 1);
                    recordMap.put(attribute, records);
                }
                records.put(entry.getKeyData(), entry);
```

Reply:
Yes, I misread your question. The index store takes care of the thread-safety, but I added the comment as you pointed out.